lib/state_machine/include/VehicleStateMachine.h

```diff
@@ -1,83 +1,70 @@
-#ifndef VEHICLE_STATE_MACHINE
-#define VEHICLE_STATE_MACHINE
+#ifndef __VehicleState_e_MACHINE__
+#define __VehicleState_e_MACHINE__
 
-/* From local systems library */
-#include "DrivetrainSystem.h"
-#include "BuzzerController.h"
+#include <etl/delegate.h>
 
-/**
- * Enum representing possible states for the vehicle's state machine.
- * 
- * STARTUP                      - Initial state. Vehicle never stays in STARTUP for more than 1 tick.
- * TRACTIVE_SYSTEM_NOT_ACTIVE   - Car stays here until the Drivetrain reports that the inverters' voltage
- *                                is above the HV threshold.
- * TRACTICE_SYSTEM_ACTIVE       - HV is on, car is waiting for brake + start button (to start car)
- * ENABLING_INVERTERS           - While in this state, the car calls the drivetrain's enable command. If
- *                                successful, (usually nearly-immediate), then car goes into WRTDS.
- * WAITING_READY_TO_DRIVE_SOUND - When entering state, buzzer is activated. After BuzzerController says
- *                                enough time has passed, we enter the next state.
- * READY_TO_DRIVE               - While in this state, pedal inputs command the drivetrain.
- */
-enum class CarState_e
-{
-    STARTUP = 0,
-    TRACTIVE_SYSTEM_NOT_ACTIVE = 1,
-    TRACTIVE_SYSTEM_ACTIVE = 2,
-    ENABLING_INVERTERS = 3,
-    WAITING_READY_TO_DRIVE_SOUND = 4,
-    READY_TO_DRIVE = 5
+enum class VehicleState_e {
+    TRACTIVE_SYSTEM_NOT_ACTIVE = 1, 
+    TRACTIVE_SYSTEM_ACTIVE = 2, 
+    WANTING_READY_TO_DRIVE = 3,
+    READY_TO_DRIVE = 4
 };
 
-/**
- * This singleton class represents the vehicle's state machine as we enable HV, enable the inverters,
- * wait for the start button to be pressed, and enter ready-to-drive mode. Aside from getters, this
- * class only has one public function, tick_state_machine(). In order to run its update logic, the
- * VehicleStateMachine uses instance data directly from other singleton classes (Buzzer, Pedals, etc)
- * and from the global structs (VCRInterfaceData_s, VCRData_s, etc.).
- */
 class VehicleStateMachine
 {
-public:
-    VehicleStateMachine(DrivetrainSystem & drivetrain_system) :
-        _current_state(CarState_e::STARTUP),
-        _drivetrain(drivetrain_system),
-        _buzzer(BuzzerController::getInstance()) {};
+    public: 
+        VehicleStateMachine(
+            etl::delegate<bool()> check_hv_over_threshold, 
```

Comment:
super minor nit: rename to _is_hv_over_threshold() to be consistent with other funcs